bundles/org.openhab.binding.sony/src/main/java/org/openhab/binding/sony/internal/scalarweb/protocols/ScalarWebAvContentProtocol.java

```diff
@@ -0,0 +1,3223 @@
+            descriptors.add(createDescriptor(createChannel(CN_SUBTITLETITLE), "String",
+                    "scalarwebavcontrolcontentsubtitletitle"));
+            descriptors.add(createDescriptor(createChannel(CN_SYNCCONTENTPRIORITY), "String",
+                    "scalarwebavcontrolcontentsynccontentpriority"));
+            descriptors.add(createDescriptor(createChannel(CN_TITLE), "String", "scalarwebavcontrolcontenttitle"));
+            descriptors.add(
+                    createDescriptor(createChannel(CN_TOTALCOUNT), "Number", "scalarwebavcontrolcontenttotalcount"));
+            descriptors.add(
+                    createDescriptor(createChannel(CN_TRIPLETSTR), "String", "scalarwebavcontrolcontenttripletstr"));
+            descriptors.add(createDescriptor(createChannel(CN_USERCONTENTFLAG), "Switch",
+                    "scalarwebavcontrolcontentusercontentflag"));
+            descriptors.add(
+                    createDescriptor(createChannel(CN_VIDEOCODEC), "String", "scalarwebavcontrolcontentvideocodec"));
+            descriptors.add(
+                    createDescriptor(createChannel(CN_VISIBILITY), "String", "scalarwebavcontrolcontentvisibility"));
+        }
+
+    }
+
+    /**
+     * Adds input status descriptors for a specific input
+     * 
+     * @param descriptors a non-null, possibly empty list of descriptors
+     * @param id a non-null, non-empty channel id
+     * @param uri a non-null, non-empty input uri
+     * @param title a non-null, non-empty input title
+     * @param apiVersion a non-null, non-empty API version
+     */
+    private void addInputStatusDescriptor(final List<ScalarWebChannelDescriptor> descriptors, final String id,
+            final String uri, final String title, final String apiVersion) {
+        Objects.requireNonNull(descriptors, "descriptors cannot be null");
+        Validate.notEmpty(id, "id cannot be empty");
+        Validate.notEmpty(uri, "uri cannot be empty");
+        Validate.notEmpty(title, "title cannot be empty");
+        Validate.notEmpty(apiVersion, "apiVersion cannot be empty");
+
+        descriptors.add(createDescriptor(createChannel(IN_URI, id, uri), "String", "scalarwebavcontrolinpstatusuri",
+                "Input " + title + " URI", null));
+
+        descriptors.add(createDescriptor(createChannel(IN_TITLE, id, uri), "String", "scalarwebavcontrolinpstatustitle",
+                "Input " + title + " Title", null));
+
+        descriptors.add(createDescriptor(createChannel(IN_CONNECTION, id, uri), "Switch",
+                "scalarwebavcontrolinpstatusconnection", "Input " + title + " Connected", uri));
+
+        descriptors.add(createDescriptor(createChannel(IN_LABEL, id, uri), "String", "scalarwebavcontrolinpstatuslabel",
+                "Input " + title + " Label", null));
+
+        descriptors.add(createDescriptor(createChannel(IN_ICON, id, uri), "String", "scalarwebavcontrolinpstatusicon",
+                "Input " + title + " Icon", null));
```

Comment:
TERM_ICON is of type "Image" but this is of type "String" ?

Reply:
Yes - term_icon is actually an image (png) where as in_icon is a type of image (ex: "meta:hdmi").  Sony - go figure.  I'll take this review item as needing to to change the label description to say ```"Input " + title + " Icon Type"